AutoImplement/Program.cs

```diff
@@ -46,9 +46,9 @@ public static class Program {
             var mainName = type.Name.Split('`')[0];
             if (mainName.StartsWith("I")) mainName = mainName.Substring(1); // strip leading 'I' from interface name

```

Comment:
Does the logic to remove a leading "I" when it makes sense need to be single sourced? It seems inconsistent between different call sites.

https://github.com/haven1433/autoimplement/blob/3a60df4da51337df10fcd0595b0122408a032d40/AutoImplement/ExtensionMethods.cs#L58

Reply:
Good point. I went ahead and single-sourced all the places that were trying to remove the leading 'I'.

While looking for other things that didn't make sense, I also moved the code that adds the using statements into the builders, to decrease how much the overall program needs to know about the code that's being generated.